### ISSUE TITLE
HTTPTupleView - Need flexible partition with attribute name in the REST query url

### ISSUE THREAD

**Issue by saberem** [IBMStreams/streamsx.inetserver#23]
Hi, we have been using the HttpTupleView in our streams projects to publish REST API for two years and we now have some use cases that require some enhancement in HTTPTupleView operator regarding the partitions.

There's a restriction in the operator that the REST API url must specify the exact amount of partitions provided in the right order. However, we'd like to query both the detailed partitions or just part of the partition as we have a use case to query for a specific client out of a client, node partition(Query attribute is not always the first attribute). And we'd like an option to disable the query to all items without a partition specified, because our root has a lot of data and query it directly caused the job to hang and crash. 

In summary, the following three functionalities, if added, will help our use case a lot:
1)Make the partition name in the url the attribute name, not the "partition" word.
2)Allow query a specified partition attribute using only that attribute name and value. 
3)Allow an optional disable of all data query without partition.

Thanks!

Cheng

**Comment by joergboe** [comment]
I will take a look on that

**Comment by joergboe** [comment]
To make sure that I understand the requirement correctly I would like to ask some questions:
Example:

the input stream:
stream<rstring clientName, rstring clientStatus, int32 amount ...>

the partitionKey parameter is:
partitionKey: "clientName", "clientStatus";

1) you would like to express the query like this:
.../0/tuples?clientName=Smith&clientStatus=gold
Which should return all tuples with attribute clientName==Smith and clientStatus==gold in the current window.
Is this correct?

2) you would also like to express a query like:
.../0/tuples?clientStatus=gold
which returns all tuples with attribute clientStatus==gold in the current window.
Is this correct?

3) You would like to have an operator parameter that forces an empty result if there is no query/partition in the current uri.
Is this correct?

4) Currently the HTTPTupleView operator supports as query parameter the values: 'partition', 'attribute', 'suppress' and 'callback'. Would you like to preserve the functionality of 'attribute', 'suppress' and 'callback'? This may induce naming conflicts when the input stream has attributes with those names.

**Comment by saberem** [comment]
Thanks @joergboe , The first three examples are all exactly what we want. For the fourth one, I don't have a preference on these as we don't use them currently and can try to avoid using these as attribute names if they are preserved.
